### PR TITLE
"Tiny" refactor

### PR DESCRIPTION
**Pull request by ChrisGora**
Made a small refactor of the Parking logic to make it free of MonoBehaviours and Transforms (in the name of easier testing!)

- All `ParkingSpace` logic is now in a `ParkingSpaceController` which is unaware of any MonoBehaviours.
- `ParkingSpaceController` timer is updated using a new `Update()` method in `ServerParkingSpaceManager`.
 - No Unity Transforms are used in the scripts. They have all been replaced by `Utils.ObjectTransform`. Make sure you use that class when passing transforms out of MonoBehaviours in the future.

### CODE REVIEW

File: ProjectFiles/Assets/Scripts/Game/Core/Parking/ParkingSpaceManager.cs [hunk 1]
```diff
@@ -13,26 +14,31 @@ namespace Game.Core.Parking {
...
             var parkingSpaceList = Object.FindObjectsOfType<ParkingSpace>();
             foreach (var parkingSpace in parkingSpaceList) {
-                parkingSpacesBySpaceID.Add(parkingSpace.SpaceID, parkingSpace);
+                parkingSpacesBySpaceID.Add(parkingSpace.Controller.SpaceID, parkingSpace.Controller);
             }
 
             DisableAllSpaces();
         }
 
+        public void TEST_ONLY_AddParkingSpace(ParkingSpaceController p) {
```
Comment (RileyEv) [inline ProjectFiles/Assets/Scripts/Game/Core/Parking/ParkingSpaceManager.cs:37]: Is that normal naming convention?

File: ProjectFiles/Packages/System.Threading.Tasks.Extensions.4.5.1/LICENSE.TXT [hunk 2]
```diff
@@ -0,0 +1,23 @@
+The MIT License (MIT)
```
Comment (RileyEv) [inline ProjectFiles/Packages/System.Threading.Tasks.Extensions.4.5.1/LICENSE.TXT]: Was this supposed to be in the gitignore?

**Review by RileyEv — COMMENTED** [review]
A few things:
- What are the InitTestScene files?
- Is some of this stuff supposed to be in the gitignore?

**Comment by RileyEv** [comment]
This certainly is the biggest tiny refactor I've ever seen. :open_mouth:

File: ProjectFiles/Assets/Scripts/Game/Core/Parking/ParkingSpaceManager.cs (hunk 1 above)
Comment (ChrisGora) [reply ProjectFiles/Assets/Scripts/Game/Core/Parking/ParkingSpaceManager.cs:37]: Yes

File: ProjectFiles/Packages/System.Threading.Tasks.Extensions.4.5.1/LICENSE.TXT (hunk 2 above)
Comment (ChrisGora) [reply ProjectFiles/Packages/System.Threading.Tasks.Extensions.4.5.1/LICENSE.TXT]: I have honestly no idea. But that library is needed for tests I think so I will remove from this branch.

**Comment by ChrisGora** [comment]
No idea what the `InitTestScene` files are doing there. They are some kind of unity testing output. Will delete.

**Comment by ChrisGora** [comment]
I removed all unused NuGet packages and the Unity Lib that's only needed for tests

**Review by RileyEv — APPROVED** [review]

**Pull request merged**